File: C_Compiler/src/IntermediateCodeGeneration/MathTuple.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package IntermediateCodeGeneration;

/**
 *
 * @author miguel
 */
public class MathTuple<T0, T1> {

    public final String acumulatedIntermediateCode;
    public final Integer ID;

    public MathTuple(String acumulatedIntermediateCode, Integer ID) {
        this.acumulatedIntermediateCode = acumulatedIntermediateCode;
        this.ID = ID;
    }

    MathTuple() {
        this.acumulatedIntermediateCode = "";
        this.ID = 0;
    }
    
    MathTuple set_acumulatedIntermediateCode(String code){
        return new MathTuple(this.acumulatedIntermediateCode + code, this.ID);
    }
    
    MathTuple set_ID(Integer ID){
        return new MathTuple(this.acumulatedIntermediateCode, ID);
    }
    
}

File: C_Compiler/src/IntermediateCodeGeneration/MathTuple.scala
package IntermediateCodeGeneration

class MathTuple {

  var acumulatedIntermediateCode: String = ""
  var ID: Integer = 0
  var operand1ID: Integer = 0

  def this(acumulatedIntermediateCode: String, ID: Integer) {
    this()
    this.acumulatedIntermediateCode = acumulatedIntermediateCode
    this.ID = ID
  }


  private[IntermediateCodeGeneration] def set_acumulatedIntermediateCode(code: String) = this.acumulatedIntermediateCode += code

  private[IntermediateCodeGeneration] def set_ID(ID: Integer) = this.ID = ID

  private[IntermediateCodeGeneration] def set_operand1ID(ID: Integer) = this.operand1ID = ID

}
